File: docs/json/combined/hs.alert.js
/**
 * @namespace hs.alert
 */
globalThis['hs.alert'] = {};

/**
 * @typedef {Object} HSAlert
 * @property {string} message - The message to display in an alert
 * @property {number} expire - How many seconds the alert should be shown for
 * @property {HSFont} font - An HSFont describing the font to use in the alert
 * @property {number} padding - How many points of padding to use in the alert
 */

/**
 * Create a new HSAlert object
 *
 * @returns {HSAlert} An HSAlert object
 */
hs.alert.newAlert = function() {};

/**
 * Show an HSAlert object
 *
 * @param {HSAlert} alert
 */
hs.alert.showAlert = function(alert) {};

/**
 * Show an alert to the user
 *
 * @param {string} message
 */
hs.alert.show = function(message) {};

File: Hammerspoon 2/Modules/hs.alert/HSAlert.swift
//
//  AlertObject.swift
//  Hammerspoon 2
//
//  Created by Chris Jones on 13/11/2025.
//

import Foundation
import JavaScriptCore
import SwiftUI

/// An object for use with hs.alert API
@objc protocol HSAlertAPI: HSTypeAPI, JSExport {
    /// The message to display in an alert
    @objc var message: String { get set }
    /// How many seconds the alert should be shown for
    @objc var expire: Int { get set }
    /// An HSFont describing the font to use in the alert
    @objc var font: HSFont { get set }
    /// How many points of padding to use in the alert
    @objc var padding: Int { get set }
}

@objc class HSAlert: NSObject, HSAlertAPI {
    @objc var typeName = "HSAlert"

    @objc var message: String = ""
    @objc var expire: Int = 5
    @objc var font: HSFont = HSFont.body()
    @objc var padding: Int = -1

    var swiftUIPadding: CGFloat? {
        guard padding >= 0 else {
            return nil
        }
        return CGFloat(padding)
    }

    var swiftUIFont: Font {
        return font.font
    }
}
